routes.py

```diff
@@ -0,0 +1,194 @@
+    user_info = auth.require_user_info(request)
+    issued_at, username = _get_provider(provider).bond.exchange_authz_code(request.args.get('oauthcode'), request.args.get('redirect_uri'), user_info)
+    return protojson.encode_message(LinkInfoResponse(issued_at=issued_at, username=username))
+
+
+@routes.route('/api/link/v1/<provider>', methods=["GET"])
+def link_info(provider):
+    user_info = auth.require_user_info(request)
+    refresh_token = _get_provider(provider).bond.get_link_info(user_info)
+    if refresh_token:
+        return protojson.encode_message(LinkInfoResponse(issued_at=refresh_token.issued_at, username=refresh_token.username))
+    else:
+        raise exceptions.NotFound("{} link does not exist".format(provider))
+
+
+@routes.route('/api/link/v1/<provider>', methods=["DELETE"])
+def delete_link(provider):
+    user_info = auth.require_user_info(request)
+    _get_provider(provider).bond.unlink_account(user_info)
+    return protojson.encode_message(message_types.VoidMessage())
+
+
+@routes.route('/api/link/v1/<provider>/accesstoken', methods=["GET"])
+def accesstoken(provider):
+    user_info = auth.require_user_info(request)
+    try:
+        access_token, expires_at = _get_provider(provider).bond.generate_access_token(user_info)
+        return protojson.encode_message(AccessTokenResponse(token=access_token, expires_at=expires_at))
+    except Bond.MissingTokenError as err:
+        # TODO: I don't like throwing and rethrowing exceptions
+        raise exceptions.BadRequest(err.message)
+
+
+@routes.route('/api/link/v1/<provider>/serviceaccount/key', methods=["GET"])
+def service_account_key(provider):
+    user_info = auth.require_user_info(request)
+    return protojson.encode_message(ServiceAccountKeyResponse(data=json.loads(
+        _get_provider(provider).fence_tvm.get_service_account_key_json(user_info))))
+
+
+@routes.route('/api/link/v1/<provider>/serviceaccount/accesstoken', methods=["GET"])
+def service_account_accesstoken(provider):
+    user_info = auth.require_user_info(request)
+    return protojson.encode_message(ServiceAccountAccessTokenResponse(token=_get_provider(provider).fence_tvm.get_service_account_access_token(user_info, request.args.getlist('scopes'))))
+
+
+@routes.route('/api/link/v1/<provider>/authorization-url', methods=["GET"])
+def authorization_url(provider):
+    authz_url = _get_provider(provider).bond.build_authz_url(request.args.getlist('scopes'), request.args.get('redirect_uri'), request.args.get('state'))
+    return protojson.encode_message((AuthorizationUrlResponse(url=authz_url)))
+
+
+@routes.route('/api/status/v1/status', methods=["GET"])
+def get_status():
+    sam_base_url = config.get('sam', 'BASE_URL')
+
+    providers = {provider_name: FenceApi(config.get(provider_name, 'FENCE_BASE_URL'))
+                 for provider_name in config.sections() if provider_name != 'sam'}
+
+    sam_api = SamApi(sam_base_url)
+    status_service = Status(sam_api, providers)
```

Comment:
Is there a way to separate out this startup stuff from the actual status query? Seems like unnecessary (although admittedly pretty trivial) work for each status call